samples/gamma-fade/main.c

```diff
@@ -0,0 +1,268 @@
+#include <stdint.h>
+#include <stdbool.h>
+#include <string.h>
+#include <math.h>
+#include <hal/video.h>
+#include <hal/debug.h>
+#include <windows.h>
+
+float remap(float in_low,  float in_high,
+            float out_low, float out_high,
+            float value)
+{
+                                 /* [in_low, in_high]       */
+    value -= in_low;             /* [0.0, in_high-in_low]   */
+    value /= in_high - in_low;   /* [0.0, 1.0]              */
+    value *= out_high - out_low; /* [0.0, out_high-out_low] */
+    value += out_low;            /* [out_low, out_high]     */
+    return value;
+}
+
+float linear(float x)
+{
+    return x;
+}
+
+/* This is also known as `smoothstep(0.0, 1.0, x)` */
+float smooth(float x)
+{
+    return x * x * (3.0f - 2.0f * x);
+}
+
+/* This is also known as `lerp(a, b, t)` */
+float mix(float a, float b, float t)
+{
+    return a * (1.0f - t) + b * t;
+}
+
+static void reset_gamma(void)
+{
+    GAMMA_RAMP_ENTRY entries[256];
+    for(int i = 0; i < 256; i++) {
+        entries[i].red   = i;
+        entries[i].green = i;
+        entries[i].blue  = i;
+    }
+    XVideoSetGammaRamp(0, entries, 256);
+}
+
+static void mix_fade(float red, float green, float blue,
+                     float t)
```

Comment:
Why the wrap? I'm not sure it helps with readability?

Reply:
This mimics the `cinematic_fade` function so the signatures and callers have the same pattern:

```c
void function([argumentGroup1,] // Fade parameters
              [argumentGroup2,] // Fade parameters
              [interpolator,] float t) // Animation control
```

*(I did have to separate the groups in the `cinematic_fade` function signature but do not separate them in the caller, to avoid vertical code, so I can see that this is not ideal either)*

Reply:
I understand the pattern, but disagree that it has readability benefits, however I don't feel strongly about it

Reply:
I also don't feel strongly; but I'd like to keep it the way it is.